persist-cli/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -108,7 +110,52 @@ public static SyntaxTree createPesistToml(String name) {
         return SyntaxTree.from(textDocument);
     }
 
-    public static HashMap<String, String> readToml(Path configPath, String name) throws BalException {
+    public static HashMap<String, String> populateConfiguration(HashMap<String, String> persistConfig, Path configPath)
+            throws BalException {
+        Path fileNamePath = configPath.getFileName();
+        try {
+            TextDocument configDocument = TextDocuments.from(Files.readString(configPath));
+            SyntaxTree syntaxTree = SyntaxTree.from(configDocument, fileNamePath.toString());
+            DocumentNode rootNote = syntaxTree.rootNode();
+            NodeList<DocumentMemberDeclarationNode> nodeList = rootNote.members();
+            for (String configKey : persistConfig.keySet()) {
+                if (Pattern.matches(PLACEHOLDER_PATTERN, persistConfig.get(configKey))) {
+                    boolean configExists = false;
+                    String[] placeHolderValues = persistConfig.get(configKey).replaceAll("\"", "").split("\\.");
+                    String relatedKey =  placeHolderValues[placeHolderValues.length - 1].replaceAll("}", "");
+                    String placeHolderTable = persistConfig.get(configKey)
+                            .replaceAll("\\$\\{", "").replaceAll("}", "").replaceAll("\\." + relatedKey, "");
+                    for (Object member : nodeList) {
+                        if (member instanceof TableNode) {
+                            TableNode node = (TableNode) member;
+                            if (node.identifier().toSourceCode().trim().equals(placeHolderTable.
+                                    replaceAll("\"", ""))) {
+                                NodeList<KeyValueNode> subNodeList = node.fields();
+                                for (KeyValueNode subMember : subNodeList) {
+                                    if (subMember.identifier().toSourceCode().trim().equals(relatedKey)) {
+                                        persistConfig.put(configKey,
+                                                subMember.value().toSourceCode().trim());
+                                        configExists = true;
+                                    }
+
+                                }
+                            }
+
+                        }
+                    }
+                    if (!configExists) {
+                        throw new BalException("Persist client related config doesn't exist in Config.toml.\n" +
+                                "You should add database configurations under <pkg name>.clients ");
```

Comment:
This validation only works if all the placeholder config doesn't exist in the Config.toml. if anyone is present this validation passes.